### PR TITLE
Feat/feature screen one domain

### PR DESCRIPTION
**Pull request by anarahari**

### CODE REVIEW

File: .idea/compiler.xml
Comment (shreyansh-mm) [inline .idea/compiler.xml:1]: This file should not be committed to VCS

File: .idea/gradle.xml
Comment (shreyansh-mm) [inline .idea/gradle.xml:1]: This file should not be committed to VCS

File: .idea/graphql-settings.xml
Comment (shreyansh-mm) [inline .idea/graphql-settings.xml:1]: This file should not be committed to VCS

File: .idea/misc.xml
Comment (shreyansh-mm) [inline .idea/misc.xml:1]: This file should not be committed to VCS

File: domain/src/main/java/com/compose/domain/paging/CharacterPagingSource.kt [hunk 1]
```diff
@@ -0,0 +1,46 @@
+package com.compose.domain.paging
+
+import androidx.paging.PagingSource
+import androidx.paging.PagingState
+import com.apollographql.apollo.ApolloClient
+import com.apollographql.apollo.api.Optional
+import com.common.graphql.GetCharactersQuery
+
+class CharacterPagingSource(
```
Comment (shreyansh-mm) [inline domain/src/main/java/com/compose/domain/paging/CharacterPagingSource.kt]: This should not be in domain

File: domain/src/main/java/com/compose/domain/repository/CharacterRepository.kt [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package com.compose.domain.repository
+
+import androidx.paging.PagingData
+import com.common.graphql.GetCharactersQuery
+import kotlinx.coroutines.flow.Flow
+
+interface CharacterRepository {
+
+    suspend fun getCharacters(): Flow<PagingData<GetCharactersQuery.Result>>
```
Comment (shreyansh-mm) [inline domain/src/main/java/com/compose/domain/repository/CharacterRepository.kt]: we should not be returning `GetCharactersQuery.Result`, we need to create our own model and map the response to our class using mapper/extension functions

**Review by shreyansh-mm — CHANGES_REQUESTED** [review]

File: domain/src/main/java/com/compose/domain/paging/CharacterPagingSource.kt (hunk 1 above)
Comment (anarahari) [reply domain/src/main/java/com/compose/domain/paging/CharacterPagingSource.kt]: done
moved it to data module

File: .idea/compiler.xml
Comment (anarahari) [reply .idea/compiler.xml:1]: done

File: .idea/gradle.xml
Comment (anarahari) [reply .idea/gradle.xml:1]: done

File: .idea/graphql-settings.xml
Comment (anarahari) [reply .idea/graphql-settings.xml:1]: done

File: .idea/misc.xml
Comment (anarahari) [reply .idea/misc.xml:1]: done

File: domain/src/main/java/com/compose/domain/usecase/GetCharactersUseCase.kt [hunk 3]
```diff
@@ -0,0 +1,13 @@
+package com.compose.domain.usecase
+
+import com.compose.domain.mapper.CharacterModel
+import com.compose.domain.repository.CharacterRepository
+import kotlinx.coroutines.flow.Flow
+import javax.inject.Inject
+
+class GetCharactersUseCase @Inject constructor(private val characterRepository: CharacterRepository) {
+
+    suspend fun getCharacters(): Flow<List<CharacterModel>> {
```
Comment (shreyansh-mm) [inline domain/src/main/java/com/compose/domain/usecase/GetCharactersUseCase.kt]: Methods of usecase are always called `invoke`

File: domain/src/main/java/com/compose/domain/mapper/CharacterModel.kt [hunk 4]
```diff
@@ -0,0 +1,10 @@
+package com.compose.domain.mapper
+
+data class CharacterModel(
+    var id: String,
+    var name: String,
+    var image: String,
+    var gender: String,
+    var species: String,
+    var status: String
```
Comment (shreyansh-mm) [inline domain/src/main/java/com/compose/domain/mapper/CharacterModel.kt]: Should be `val` not `var`

**Review by shreyansh-mm — CHANGES_REQUESTED** [review]

**Pull request merged**